docs/DeveloperGuide.md

```diff
@@ -2,37 +2,381 @@
+intake, cardio sessions, personal bests, or goals. It maintains a List<Log> and provides methods to add, delete, update,
+list, and search through these logs.
+
+- The abstract Log class is the superclass for all log types.
+- Subclasses like Meal, Workout, Water, Cardio, PersonalBest, and Goals extend Log and store specific data for each log type.
+- LogList has a one-to-many association with Log, represented by the contains relationship.
+- The Goals class follows a singleton pattern via getInstance() and stores user-defined fitness goals.
+
+This design promotes extensibility and encapsulation, allowing new log types to be added easily while maintaining a consistent interface.
+
+### Storage
+The Storage class manages HealthBud log persistence by reading and writing to a designated text file. It creates necessary directories and files, loads logs by parsing each line into specific types (Meal, Workout, etc.), and distributes them into corresponding LogLists. It also appends and rewrites logs using formatted string representations while handling errors gracefully.
+![Storage_Class_Diagram](images/StorageCD.png)
+
+### Data
+`//TODO: include command class diagram here`
+
+
+# Implementation
+
+## Add Log Command
+`//TODO: include SD here`
+
+## Delete Log Command
+The delete log feature allows users to remove a log by its index from the application's log list. This feature is handled by the `DeleteCommand` class, which performs validation, deletion, and error handling.
+
+### 1. User Input: <br>
+   - The user enters the delete command followed by the log's index (e.g., `delete meal 3`).
+
+
+### 2. Command Parsing: <br>
+   - The Parser converts the input into a `DeleteCommand` object, adjusting the index to match the list’s 0-based indexing. This process is handled by the `ParserManager` and the `DeleteParser.`
+
+
+### 3. The `DeleteCommand` executes as follows: <br>
+   - Validation: Verifies whether the specified index is valid and corresponds to an existing meal log.
+
+   - Error Handling: If the index is invalid, an error message is returned to the user.
+
+   - Deletion: If the index is valid, the command retrieves the log’s details, removes the log from the Logs list, and generates a success message.
+
+
+### 4. How the feature is implemented: <br>
+   - The deletion functionality is handled by the DeleteCommand class. It validates the user-provided index, adjusts it to match the 0-based indexing of the log list, and performs the deletion on the LogList object. This keeps the deletion logic isolated, making it easier to maintain and test.
+
+
+### 5. Why it is implemented that way: <br>
+    - Using a dedicated command class follows the Command Pattern, which separates concerns effectively. Isolating deletion logic into its own class adheres to the Single Responsibility Principle, simplifying debugging and future enhancements without impacting other parts of the system.
```

Comment:
The scrolling format is a bit difficult to read, could be normal paragraph instead, but clear explanation